app/src/main/java/com/kickstarter/ui/activities/PaymentMethodsSettingsActivity.kt

```diff
@@ -0,0 +1,93 @@
+package com.kickstarter.ui.activities
+
+import UserPaymentsQuery
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AlertDialog
+import android.support.v7.widget.LinearLayoutManager
+import com.kickstarter.R
+import com.kickstarter.extensions.showConfirmationSnackbar
+import com.kickstarter.extensions.showErrorSnackbar
+import com.kickstarter.libs.ActivityRequestCodes
+import com.kickstarter.libs.BaseActivity
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel
+import com.kickstarter.libs.utils.ViewUtils
+import com.kickstarter.ui.adapters.PaymentMethodsAdapter
+import com.kickstarter.viewmodels.PaymentMethodsViewModel
+import kotlinx.android.synthetic.main.activity_settings_payment_methods.*
+import kotlinx.android.synthetic.main.payment_methods_toolbar.*
+import rx.android.schedulers.AndroidSchedulers
+
+@RequiresActivityViewModel(PaymentMethodsViewModel.ViewModel::class)
+class PaymentMethodsSettingsActivity : BaseActivity<PaymentMethodsViewModel.ViewModel>() {
+
+    private lateinit var adapter: PaymentMethodsAdapter
+    private var showDeleteCardDialog: AlertDialog? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_settings_payment_methods)
+
+        setUpRecyclerView()
+
+        this.viewModel.outputs.cards()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { setCards(it) }
+
+        this.viewModel.outputs.error()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { showErrorSnackbar(payment_methods_toolbar, it) }
+
+        this.viewModel.outputs.progressBarIsVisible()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { ViewUtils.setGone(progress_bar, !it) }
+
+        this.viewModel.outputs.showDeleteCardDialog()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { lazyDeleteCardConfirmationDialog().show() }
+
+        this.viewModel.success()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { showConfirmationSnackbar(payment_methods_toolbar, R.string.Got_it_your_changes_have_been_saved) }
```

Comment:
Curious why the toolbar layout opposed to the activity's layout?

Reply:
I usually like reusing an existing view as the anchor instead of creating a new id that only gets used to show Snackbars. Definitely just one of my many style quirks O_<